common/io/src/hts_fasta_parser.cpp

```diff
@@ -60,29 +60,37 @@ int32_t FastaParserHTS::get_num_seqences() const
     return num_seqequences_;
 }
 
-FastaSequence FastaParserHTS::get_sequence_by_id(int32_t i) const
```

Comment:
Why can't this method be `const`? Mutex? In that case you can use `mutable`

Reply:
This feels like more a matter of opinion :)

Reply:
Partially, but const-correctness is a very important thing and should be done right whenever possible. Currently we don't mind passing a non-const `FastaParser` to `Index`, but this could change in the future. Also, using ``mutable` for mutex is a rather common practice in this case in C++, it's not considered hacky.
@ahehn-nv, what do you think?

Reply:
I agree with @mimaric. If you look at the abstract logic of the class, this method leaves the object in the same state, except for an internal implementation detail. From the outside the object is unchanged.

I would expect a `const` method to be thread-safe though - but that's exactly what this detail is supposed to do.

Reply:
@ahehn-nv @mimaric reverted the methods to be `const` again. Reasons above convinced me that `mutable` is the way to go. abstract API indeed remains const

Reply:
Looks good, marking as resolved